api/omnitransaction.py

```diff
@@ -0,0 +1,711 @@
+            #DEBUG print [obpacket, len(obpacket), shaaddress, len(shaaddress)]
+            datapacket = ''
+            for i in range(len(obpacket)):
+                if obpacket[i] == shaaddress[i]:
+                    datapacket = datapacket + '0'
+                else:
+                    bin_ob = int('0x' + obpacket[i], 16)
+                    bin_sha = int('0x' + shaaddress[i], 16)
+                    xored = hex(bin_ob ^ bin_sha)[2:].upper()
+                    datapacket = datapacket + xored
+            plaintext_packets.append(datapacket)
+        
+        #check the packet is formed correctly by comparing it to the input
+        final_packets = []
+        for i in range(len(plaintext_packets)):
+            orig = packets[i]
+            if orig.upper() != plaintext_packets[i]:
+                print ['packet did not come out right', orig, plaintext_packets[i] ]
+            else:
+                final_packets.append(obfuscated_packets[i][0])
+        
+        #DEBUG print plaintext_packets, obfuscation_packets,final_packets
+        
+        #add key identifier and ecdsa byte to new mastercoin data key
+        for i in range(len(final_packets)):
+            obfuscated = '02' + final_packets[i] + "00" 
+            #DEBUG print [obfuscated, len(obfuscated)]
+            invalid = True
+            while invalid:
+                obfuscated_randbyte = obfuscated[:-2] + hex(random.randint(0,255))[2:].rjust(2,"0").upper()
+                #set the last byte to something random in case we generated an invalid pubkey
+                potential_data_address = pybitcointools.pubkey_to_address(obfuscated_randbyte, self.magicbyte)
+                
+                if bool(self.conn.validateaddress(potential_data_address).isvalid):
+                    final_packets[i] = obfuscated_randbyte
+                    invalid = False
+            #make sure the public key is valid using pybitcointools, if not, regenerate 
+            #the last byte of the key and try again
+        
+        #DEBUG print final_packets 
+        return [final_packets,total_packets,total_outs]
+        
+    def __build_transaction(self, miner_fee_btc, pubkey,final_packets, total_packets, total_outs, from_address, to_address=None):
+        print "build_transaction", miner_fee_btc, pubkey,final_packets, total_packets, total_outs, from_address, to_address
+        print 'pubkey', pubkey, len(pubkey) 
+        if len(pubkey) < 100:
+          print "Compressed Key, using hexspace 21"
+          HEXSPACE_FIRST='21'
+        else:
+          HEXSPACE_FIRST='41'
+
+        #calculate fees
+        miner_fee = Decimal(miner_fee_btc)
+        if to_address==None or to_address==from_address:
+             #change goes to sender/receiver
+            print "Single extra fee calculation"  
+            fee_total = Decimal(miner_fee) + Decimal(0.00005757*total_packets+0.00005757*total_outs) + Decimal(0.00005757)  #exodus output is last
```

Comment:
Good point, do you mean extract them into a constant? or is there a better way to get those values from omnicore?